stoobly_agent/app/cli/helpers/endpoints_import_service.py

```diff
@@ -1,123 +1,153 @@
-import requests
 import json
+import requests
+from typing import TypedDict
 
+from .endpoints_import_context import EndpointsImportContext
+from stoobly_agent.app.models.types import ENDPOINT_COMPONENT_NAMES
+from stoobly_agent.lib.api.body_param_names_resource import BodyParamNamesResource
+from stoobly_agent.lib.api.endpoints_resource import EndpointsResource
+from stoobly_agent.lib.api.header_names_resource import HeaderNamesResource
+from stoobly_agent.lib.api.interfaces.endpoints import BodyParamName
 from stoobly_agent.lib.api.interfaces.endpoints import EndpointShowResponse
 from stoobly_agent.lib.api.interfaces.endpoints import RequestComponentName
-from stoobly_agent.lib.api.interfaces.endpoints import BodyParamName
-from stoobly_agent.lib.api.endpoints_resource import EndpointsResource
-
-from .endpoints_import_context import EndpointsImportContext
-
-def import_all_endpoints(context: EndpointsImportContext):
+from stoobly_agent.lib.api.query_param_names_resource import QueryParamNamesResource
+from stoobly_agent.lib.api.response_header_names_resource import ResponseHeaderNamesResource
+from stoobly_agent.lib.api.response_param_names_resource import ResponseParamNamesResource
+
+class PendingImportArgs(TypedDict):
+    project_id: int
+    endpoint_id: int
+    component_name: str
+    component: RequestComponentName
+    component_resource_class: EndpointsResource
+
+def import_endpoints(context: EndpointsImportContext):
+    uploaded_endpoints = []
     for endpoint in context.endpoints:
         for endpoint_handler in context.endpoint_handlers:
             endpoint_handler(endpoint)
 
-        import_endpoint_and_component_name_resources(context, endpoint)
-
-def import_endpoint_and_component_name_resources(context: EndpointsImportContext, endpoint: EndpointShowResponse):
-    res = import_endpoint(context, endpoint)
+        res = import_endpoint_and_component_name(context, endpoint, uploaded_endpoints)
+        if not res.ok:
```

Comment:
API will return 409 when an endpoint has already been created. In this case `res.ok` won't be `True`, but we also don't want to cleanup the endpoint.